### PR TITLE
Add 'Reducing the Memory Footprint' page

### PR DESCRIPTION
**Pull request by maxkernbach**
https://github.com/arangodb/planning/issues/3728

### CODE REVIEW

**Comment by Simran-B** [comment]
@sleto-it Not sure about putting it into the Troubleshooting chapter yet... should it rather be located under Administration or perhaps Tutorials even?

File: 3.5/troubleshooting-reduce-memory-footprint.md [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+-------------------
+
+RocksDB writes into
+[memory buffers mapped to on-disk blocks](https://github.com/facebook/rocksdb/wiki/Write-Buffer-Manager)
+first. At some point, the memory buffers will be full and have to be written
+to disk. In order to support high write loads, RocksDB might open a lot of these
+memory buffers.
+
+Under normal write load, the write buffers will use less than 1 GByte of memory.
+If you are tight on memory, or your usage pattern doesn’t require this, you can
```
Comment (sleto-it) [inline 3.5/troubleshooting-reduce-memory-footprint.md]: doesn't --> does not

File: 3.5/troubleshooting-reduce-memory-footprint.md [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+The main deployment target for ArangoDB is Linux. As you’ve learned above
+ArangoDB and its innards work a lot with memory. Thus its vital to know how
+ArangoDB and the Linux kernel interact on that matter. The linux kernel offers
+several modes of how it will manage memory. You can influence this via the proc
+filesystem, the file `/etc/sysctl.conf` or a file in `/etc/sysctl.conf.d/` which
+your system will apply to the kernel settings at boot time. The settings as
+named below are intended for the sysctl infrastructure, meaning that they map
+to the `proc` filesystem as `/proc/sys/vm/overcommit_memory`.
+
+After studying the kernel documentation our former recommendation was to set
```
Comment (sleto-it) [inline 3.5/troubleshooting-reduce-memory-footprint.md]: this is not good imho - this is correct for a blog post, but not for the guide - my opinion

File: 3.5/troubleshooting-reduce-memory-footprint.md [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+memory mappings to each process, which defaults to 65530 on many Linux
+environments.
+
+Another issue when running jemalloc with `vm.overcommit_memory` set to **2** is
+that for some workloads the amount of memory that the Linux kernel tracks as
+“*committed memory*” also grows over time and never decreases. Eventually,
+*arangod* may not get any more memory simply because it reaches the configured
+overcommit limit (physical RAM * `overcommit_ratio` + swap space).
+
+The solution is to modify the value of `vm.overcommit_memory` from **2** to
```
Comment (sleto-it) [inline 3.5/troubleshooting-reduce-memory-footprint.md]: no link to the section of the guide where we also talk about the topic https://www.arangodb.com/docs/3.4/installation-linux-osconfiguration.html#over-commit-memory

File: 3.5/troubleshooting-reduce-memory-footprint.md [hunk 4]
```diff
@@ -0,0 +1,353 @@
...
+[upstream documentation](https://github.com/facebook/rocksdb/wiki/Write-Buffer-Manager):
+
+> […] a flush will be triggered […] if total mutable memtable size exceeds 90%
+> of the limit. If the actual memory is over the limit, more aggressive flush
+> may also be triggered even if total mutable memtable size is below 90%.
+
+Since we only measured the disk I/O bytes, we don’t see that the document save
+operations also doubled in request time.
+
+Summary
```
Comment (sleto-it) [inline 3.5/troubleshooting-reduce-memory-footprint.md]: again blog post or tutorial style

**Comment by sleto-it** [comment]
Hi @Simran-B I do not think it can be located under Administration, the way it is written. Needs some changes first
Administration is a chapter of the reference manual - the style does not fit, should be changed first

A tutorial probably okay

We should link other section of the documentation (e.g. https://www.arangodb.com/docs/3.4/installation-linux-osconfiguration.html#over-commit-memory) otherwise it seems something included in the doc and unlinked to other - I think we should avoid such situation

Hope the above helps

File: 3.5/troubleshooting-reduce-memory-footprint.md (hunk 2 above)
Comment (Simran-B) [reply 3.5/troubleshooting-reduce-memory-footprint.md]: shortened the sentence by a big deal

File: 3.5/troubleshooting-reduce-memory-footprint.md (hunk 4 above)
Comment (Simran-B) [reply 3.5/troubleshooting-reduce-memory-footprint.md]: moved to the top to act as an introduction

**Comment by Simran-B** [comment]
@sleto-it Changed to be added as tutorial, use the summary as introduction, reference the linux os conf page, reworded a few sentences:

https://github.com/arangodb/docs/pull/25/commits/7ba0d8932a5b949d4a11ca603552c775f4b971c9

**Pull request merged**